File: Day04/Palindrome.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        temp = str(x)
        a = ''
        b = ''
        for i in temp:
            a += i
        for j in range(len(temp)-1, -1, -1):
            b += temp[j]
        if a == b:
            return True
        else:
            return False

def main():
    solution = Solution()
    test_cases = [121, -121, 10, 545, 12321, 278]
    
    for num in test_cases:
        print(f"{num} is palindrome? {solution.isPalindrome(num)}")

if __name__ == "__main__":
    main()

File: Day04/Palindrome.java
class Palindrome {
    public boolean isPalindrome(int x) {
        String temp = Integer.toString(x);
        String a = "";
        String b = "";
        int len = temp.length();
        for (int i = 0; i < len; i++) {
            a += temp.charAt(i);
        }
        for (int j = len - 1; j >= 0; j--) {
            b += temp.charAt(j);
        }
        return a.equals(b);
    }

    public static void main(String[] args) {
        PerfectNo solution = new PerfectNo();
        int[] testCases = {121, -121, 10, 545, 12321};
        
        for (int num : testCases) {
            System.out.println(num + " is palindrome? " + solution.isPalindrome(num));
        }
    }
}